lib/id3c/cli/command/etl/__init__.py

```diff
@@ -357,19 +357,70 @@ def upsert_location(db: DatabaseSession,
     else:
         hierarchy = hierarchy + "," + location_hierarchy
 
-    location = db.fetch_row("""
-        insert into warehouse.location (scale, identifier, hierarchy)
-        values (%s, %s, %s)
+    data = {
+        "scale": scale,
+        "identifier": identifier,
+        "hierarchy": hierarchy
+    }
+
+    with db.cursor() as cursor:
+        cursor.execute("""
+            select
+                location_id as id,
+                scale,
+                identifier,
+                hierarchy,
+                (location.hierarchy) != (%(hierarchy)s) as data_changed
```

Comment:
We'll want to cast the incoming value to `hstore` data type for the comparison. The extra parenthesis aren't necessary.